### DISCUSSION TITLE
NPC Drop Custom Sounds - plugin suggestion

### DISCUSSION THREAD

**Discussion by BloodhoundH** [runelite/runelite#15010] [Ideas]
### NPC Drop Custom Sounds
I've been using the [Qolting ](https://github.com/ArtsicleOfficial/Qolting/tree/83e95d2c2cc77c1d0eadefaa9ea8bbec62eef5b5)plugin made by ArtsicleOfficial while killing vyres and there is a fantastic part of this plugin that allows you to play custom sound (.wav) files for various NPC drops, such as when you receive a Blood Shard, an item drop over a certain value etc. The plugin is mainly quality of life for killing vyres and as such doesn’t serve much purpose outside of Darkmeyer.

The sound blasters part of the Qolting plugin is an amazing function that I feel deserves to be experienced by more players, as a standalone plugin to allow runelite users to add an extra dimension to getting a high value drop (or saught after drops such as clue scrolls) by playing custom sound files they set themselves.

The Ground Items plugin already has some great ways to highlight valuable items dropped by NPC's using loot beams. While a visual cue is a great sight to see when you get such lucrative drops, an audio cue adds an extra layer to the experience of receiving the drop you’re looking for.

### Plugin Details
**Plugin Description:** Lets you play a custom sound for valuable & whitelisted NPC drops.

_Settings:_
<img src="https://i.imgur.com/rcH7UK3.png">

**Valuable Threshold** - (e.g 5,000,000) (Threshold in traded value for NPC drops to be considered valuable. This will play the sound file specified in ‘Valuable Blaster’ setting.)
**Drop Whitelist** - (Add items to the 'Drop Whitelist’ that you want to play the sound file specified in ‘Whitelist Blaster’ setting. e.g untradable drops such as Clue Scrolls, Ancient Shards).
**Valuable Blaster** - (Place file in '%userprofile%\.runelite\npcdropsounds\valuable.wav’) [Enable/Disable]
**Whitelist Blaster** - (Place file in '%userprofile%\.runelite\npcdropsounds\whitelist.wav’) [Enable/Disable]

The Qolting plugin already has a way to identify if an NPC dropped an item over a certain value, where it scans NPC drops and if the traded value of the item is above your threshold set then it will play your custom sound .wav file. I have checked with ArtsicleOfficial to see if they had any plans on making a standalone npc drop sound plugin but they do not have interest in doing so at present, which is why I’m posting here as a potential plugin suggestion should it take anyone’s interest! 😊

_Some other outside of scope ideas for the plugin:_
Clue Scroll Treasure Value Threshold (e.g 1,000,000) (Threshold in clue scroll treasure value to be considered valuable.) – Play a sound when a clue scroll is over a threshold specified.

**Comment by Enriath** [comment]
> The Ground Items plugin already has some great ways to highlight valuable items dropped by NPC's using loot beams. While a visual cue is a great sight to see when you get such lucrative drops, an audio cue adds an extra layer to the experience of receiving the drop you’re looking for.

Just gonna mention that there already is the option for audio through the notification system.

**Reply by BloodhoundH** [reply]
That's a fair point and the runelite notification system custom sound does work to a certain degree of what I'm hoping for. The constraint I'm finding is if I'm wanting to set a sound clip say of 10's of [this youtube video](https://www.youtube.com/watch?v=HMuYfScGpbE) just for insane value drops over 5m, then it's going to play everytime any other plugin utlises the runelite notification system. So low prayer and out of combat notifications will trigger the sound etc.

**Reply by Enriath** [reply]
Yeah. Plugin-specific custom notification sounds are something that's been asked for quite a lot over the years, with the main issue being doing it in a way that isn't arse. I think #10404 is the closest we've gotten to it, though there's still an issue of UI/UX.
